src/components/modal/createModalLauncher.tsx

```diff
@@ -0,0 +1,35 @@
+import * as React from 'react';
+import { Modal, ModalProps } from '@patternfly/react-core';
+
+export type ModalComponentProps = Omit<ModalProps, 'children' | 'isOpen' | 'appendTo'> & {
+  ref?: React.LegacyRef<Modal>;
+};
+
+export type ModalLauncher = (onClose: () => void) => React.ReactElement;
+
+export type CreateModalLauncher = <P extends ModalComponentProps>(
+  Component: React.ComponentType<P>,
+  modalConfig?: ModalComponentProps & {
+    'data-testid': string;
+  },
+) => (componentProps?: P) => ModalLauncher;
```

Comment:
Isn't `componentProps` just for passing data to actual component being rendered inside modal like `Component` data?

Reply:
@rohitkrai03  yes, it is
do you want me to name this to something else ? may be `componentData` ?